projects/packages/search/src/dashboard/components/pages/sections/plan-usage-section.jsx

```diff
@@ -24,6 +24,207 @@ const usageInfoFromAPIData = apiData => {
+		messageOverLimitThree,
+		messageNearLimitTwo, // not used
+		messageNearLimitThree, // not used
+	][ index ];
+	// Possible CTA messages here.
+	const ctaUpgradeToAvoid = __(
+		'Upgrade now to increase your monthly request limit and avoid interruption.',
+		'jetpack-search-pkg'
+	);
+	const ctaUpgradeToContinue = __(
+		'Upgrade now to contineu using Jetpack Search.',
+		'jetpack-search-pkg'
+	);
+	// Pick the appropriate CTA.
+	let cta = ctaUpgradeToAvoid;
+	if ( apiData.currentUsage.months_over_plan_requests_limit >= 3 ) {
+		cta = ctaUpgradeToContinue;
+	}
+
+	return { description: message, cta: cta };
+};
+
+const upgradeMessageNoOverage = apiData => {
+	// Always returns a valid message.
+	const message = __(
+		'Do you want to increase your site records and search requests?.',
+		'jetpack-search-pkg'
+	);
+	const cta = __( 'Upgrade now and avoid any future interruption!', 'jetpack-search-pkg' );
+	return { description: message, cta: cta };
+};
+
+const upgradeMessageBoth = apiData => {
+	// Return a valid message or null.
+	const isBothScenario =
+		apiData.currentUsage.upgrade_reason.records && apiData.currentUsage.upgrade_reason.requests;
+	if ( ! isBothScenario ) {
+		return null;
+	}
+	// Determine upgrade message.
+	const messageNearLimit = __(
+		'You’re close to exceeding the number of site records and search requests available for the free plan.',
+		'jetpack-search-pkg'
+	);
+	const messageOverLimit = __(
+		'You’ve exceeded the number of site records and search requests available for the free plan.',
+		'jetpack-search-pkg'
+	);
+	// If the "months over" is zero, use the near messaging.
+	let message = messageOverLimit;
+	if ( apiData.currentUsage.months_over_plan_records_limit === 0 ) {
+		message = messageNearLimit;
+	}
+	// Add the shared CTA.
+	const sharedCTA = __(
+		'Upgrade now to increase your limits and avoid interruption!',
+		'jetpack-search-pkg'
+	);
+	return { description: message, cta: sharedCTA };
+};
+
+const upgradeMessageFromAPIData = apiData => {
+	// What's the data we're working with?
+	// apiData.currentUsage.must_upgrade
+	// apiData.currentUsage.upgrade_reason.records
+	// apiData.currentUsage.upgrade_reason.requests
+	// apiData.currentUsage.months_over_plan_records_limit
+	// apiData.currentUsage.months_over_plan_requests_limit
+	if ( ! apiData.currentUsage.must_upgrade ) {
+		return null;
+	}
+	// Handle both case.
+	let message = upgradeMessageBoth( apiData );
+	if ( message ) {
+		return message;
+	}
+	// Handle Records overages.
+	message = upgradeMessageRecords( apiData );
+	if ( message ) {
+		return message;
+	}
+	// Handle Requests overages.
+	message = upgradeMessageRequests( apiData );
+	if ( message ) {
+		return message;
+	}
+	// Handle the default case.
+	return upgradeMessageNoOverage();
+};
+
+// TODO: Remove this if no longer needed.
+// Currently not called. Not removing yet, pending review of new CTA logic (which feels messy).
+// eslint-disable-next-line no-unused-vars
```

Comment:
I understand the desire to preserve existing code for a future refactor. However, I'd like for us to avoid adding linting overrides to preserve currently unused code; more code volume makes the code base significantly harder to learn.

In our case, it's likely that we may not return to this codebase in the near future.